src/masterless.py

```diff
@@ -0,0 +1,60 @@
+from fabric.api import local, task
+from decorators import requires_project, requires_aws_stack, echo_output
+from buildercore import bootstrap, config, core
+from buildercore.utils import ensure
+import logging
+from functools import wraps
+LOG = logging.getLogger(__name__)
+
+def requires_master_server_access(fn):
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+        import master
+        ensure(master.server_access(), "this command requires access to the master server. you don't have it.")
```

Comment:
can this be cached inside the current process? If multiple tasks each requiring access are involved, no reason to check it more than once, for better performance